Modules/Microsoft365DSC/Modules/M365DSCPermissions.psm1

```diff
@@ -1714,7 +1714,7 @@ function Update-M365DSCAzureAdApplication
                     if ($CreateSelfSignedCertificate)
                     {
                         Write-LogEntry '    CreateSelfSignedCertificate specified, generating new Self Signed Certificate'
-                        $cerCert = New-SelfSignedCertificate -CertStoreLocation 'Cert:\CurrentUser\My' `
+                        $cerCert = New-SelfSignedCertificate -CertStoreLocation 'Cert:\LocalMachine\My' `
```

Comment:
This part depends on what the intent is. If the user wants to do an Export, than the CurrentUser store is where the certificate should be placed. If it's a logic path using the LCM (e.g., Start/Test-DSCConfiguration) then LocalMachine is required.

Reply:
but currently it does neither an export-pfx nor places it in a store for subsequent use. So as a new user, I cannot use this scriptlet to configure a service principal for me.